dask/tests/test_tokenize.py

```diff
@@ -1271,6 +1271,7 @@ def test_tokenize_random_functions(module):
     """random.random() and other methods of the global random state do not compare as
     equal to themselves after a pickle roundtrip"""
     module = eval(module)
+    module.seed(2)
```

Comment:
Manually setting the random seed here gets this test passing. If we don't do this, then this test fails against NumPy 2 with this tokenization mismatch error:

```
__________________ test_tokenize_random_functions[np.random] ___________________
[gw0] linux -- Python 3.12.4 /home/runner/miniconda3/envs/test-environment/bin/python3.12

module = <module 'numpy.random' from '/home/runner/miniconda3/envs/test-environment/lib/python3.12/site-packages/numpy/random/__init__.py'>

    @pytest.mark.parametrize(
        "module",
        ["random", pytest.param("np.random", marks=pytest.mark.skipif("not np"))],
    )
    def test_tokenize_random_functions(module):
        """random.random() and other methods of the global random state do not compare as
        equal to themselves after a pickle roundtrip"""
        module = eval(module)
    
        a = module.random
        b = pickle.loads(pickle.dumps(a))
>       assert check_tokenize(a) == check_tokenize(b)
E       AssertionError: assert '5a758251ee05...fadb5e734ae74' == 'f8c814ecbf02...12210d58c137e'
E         
E         - f8c814ecbf0209a0c4812210d58c137e
E         + 5a758251ee0551802acfadb5e734ae74

dask/tests/test_tokenize.py:1277: AssertionError
```

@fjetter @phofl  @hendrikmakait do any of you have more context here?

Reply:
not really no, but this sounds like an acceptable solution